codec/codec-fury/src/main/java/com/alipay/sofa/rpc/codec/fury/FurySerializer.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alipay.sofa.rpc.codec.fury;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import com.alipay.sofa.rpc.codec.AbstractSerializer;
+import com.alipay.sofa.rpc.common.RemotingConstants;
+import com.alipay.sofa.rpc.common.struct.ConcurrentHashSet;
+import com.alipay.sofa.rpc.common.utils.CodecUtils;
+import com.alipay.sofa.rpc.common.utils.StringUtils;
+import com.alipay.sofa.rpc.config.ConfigUniqueNameGenerator;
+import com.alipay.sofa.rpc.context.RpcInvokeContext;
+import com.alipay.sofa.rpc.core.exception.SofaRpcException;
+import com.alipay.sofa.rpc.core.request.SofaRequest;
+import com.alipay.sofa.rpc.core.response.SofaResponse;
+import com.alipay.sofa.rpc.ext.Extension;
+import com.alipay.sofa.rpc.transport.AbstractByteBuf;
+import com.alipay.sofa.rpc.transport.ByteArrayWrapperByteBuf;
+
+import io.fury.Fury;
+import io.fury.Language;
+import io.fury.serializer.CompatibleMode;
+
+/**
+ * @author lipan
+ */
+@Extension(value = "fury", code = 20)
+public class FurySerializer extends AbstractSerializer {
+
+    private final FuryHelper    furyHelper  = new FuryHelper();
+    private final Set<Class<?>> registerSet = new ConcurrentHashSet<>();
+    private final Fury          fury;
+
+    public FurySerializer() {
+        fury = Fury.builder()
```

Comment:
If a thread-safe fury is constructed first, fury will no longer be able to register any class. 
There's no idea. I'll try to: use a `for` ,when there is a new class, put it in set, build a new fury, and register the class in set again.it looks stupid

Reply:
how would u implement such registration logic this time? just open an api for user to register?

Reply:
This question, I think, can be discussed together with the above question.
My original design was to use a registerSet and when a new class is not in the set, register it with fury and add it to the registerSet.
However, if you want to use ThreadLocalFury, I will provide a default way to not register, because fury can no longer accept any class registration after becoming ThreadLocalFury, but even so, in most cases, it will work.
Then, I'll provide another constructor, which is that only the first registered class will work

```java
    private final ThreadLocalFury fury;

    public FurySerializer() {
        fury = new ThreadLocalFury(classLoader -> {

            Fury f = Fury.builder().withLanguage(Language.JAVA)
                .withRefTracking(false)
                .withCodegen(true)
                .withNumberCompressed(true)
                .withCompatibleMode(CompatibleMode.COMPATIBLE)
                .requireClassRegistration(false)
                .withClassLoader(classLoader)
                .withAsyncCompilationEnabled(true)
                .build();
            return f;
        });
    }

    public FurySerializer(List<Class<?>> classList) {
        fury = new ThreadLocalFury(classLoader -> {

            Fury f = Fury.builder().withLanguage(Language.JAVA)
                .withRefTracking(false)
                .withCodegen(true)
                .withNumberCompressed(true)
                .withCompatibleMode(CompatibleMode.COMPATIBLE)
                .requireClassRegistration(true)
                .withClassLoader(classLoader)
                .withAsyncCompilationEnabled(true)
                .build();
            for (Class<?> clazz : classList) {
                f.register(clazz);
            }
            return f;
        });
    }

```